dev/tasks/r/github.packages.yml

```diff
@@ -309,12 +309,22 @@ jobs:
           curl -s \
             https://raw.githubusercontent.com/{{ arrow.github_repo }}/{{ arrow.head }}/ci/scripts/install_sccache.sh | \
             bash -s unknown-linux-musl /usr/local/bin
-      - run: sudo apt update && sudo apt install libcurl4-openssl-dev
-      - name: Prepare PKG_CONFIG_PATH for Homebrew
+      - name: Install curl and its dependencies for static link
         run: |
-          # zstd is installed by Homebrew on GitHub Actions.
-          echo "PKG_CONFIG_PATH=$(brew --prefix)/lib/pkgconfig${PKG_CONFIG_PATH:+:${PKG_CONFIG_PATH}}" \
-            >> "$GITHUB_ENV"
+          sudo apt update
+          sudo apt install -y \
+            libbrotli-dev \
+            libcurl4-openssl-dev \
+            libidn2-dev \

```

Comment:
I don't understand, why do we need all of these extra dependencies now? If they're dependencies of libcurl, why isn't installing libcurl enough?